### PR TITLE
libobs: Fix flip calculations in bounding box

### PR DESCRIPTION
**Pull request by Penwy**
### Description

This fixes flipping behaviour when positional alignement inside the bounding box is other than centered, so that the item gets flipped in-place within the bounding box, instead of being thrown out of the bounding box.

### Motivation and Context

Currently, when a sceneitem set in a bounding box with an alignment other than "center" gets flipped, it is thrown out of the bounding box, which does not feel like the expected behaviour of a bounding box.

### How Has This Been Tested?

Ubuntu 22.04
Tested various combinations of positional alignement and bounding box types, checked that all correctly flipped the item in-place on x and y.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by norihiro** [comment]
The description has a 75-character line, where the limit is 72 characters.

File: libobs/obs-scene.c [hunk 1]
```diff
@@ -401,13 +401,17 @@ static void calculate_bounds_data(struct obs_scene_item *item,
 
 	width = (float)(*cx) * scale->x;
 	height = (float)(*cy) * scale->y;
-	width_diff = item->bounds.x - width;
-	height_diff = item->bounds.y - height;
+	// Disregards flip when calculating size diff
```
Comment (cg2121) [inline libobs/obs-scene.c:404]: In c files it is suggested to /* Comment */ styled comments, instead of //

File: libobs/obs-scene.c [hunk 2]
```diff
@@ -401,13 +401,17 @@ static void calculate_bounds_data(struct obs_scene_item *item,
...
+	width_diff = item->bounds.x - fabsf(width);
+	height_diff = item->bounds.y - fabsf(height);
 	*cx = (uint32_t)item->bounds.x;
 	*cy = (uint32_t)item->bounds.y;
 
 	add_alignment(origin, item->bounds_align, (int)-width_diff,
 		      (int)-height_diff);
+	// Makes the item stay in-place in the box if flipped
```
Comment (cg2121) [inline libobs/obs-scene.c:412]: Same here

File: libobs/obs-scene.c (hunk 1 above)
Comment (RytoEX) [reply libobs/obs-scene.c:404]: That is only required for multiline comments. For single line comments, there is no set rule for `//` vs. `/* */`.

**Pull request merged**